frontend/pages/admin/IntegrationsPage/cards/CertificateAuthorities/components/HydrantForm/helpers.ts

```diff
@@ -0,0 +1,149 @@
+import { ICertificatesIntegrationHydrant } from "interfaces/integration";
+
+import valid_url from "components/forms/validators/valid_url";
+
+import { IHydrantFormData } from "./HydrantForm";
+
+// TODO: create a validator abstraction for this and the other form validation files
+
+export interface IHydrantFormValidation {
+  isValid: boolean;
+  name?: { isValid: boolean; message?: string };
+  url?: { isValid: boolean; message?: string };
+  clientId?: {
+    isValid: boolean;
+    message?: string;
+  };
+  clientSecret?: { isValid: boolean; message?: string };
+}
+
+type IMessageFunc = (formData: IHydrantFormData) => string;
+type IValidationMessage = string | IMessageFunc;
+type IFormValidationKey = keyof Omit<IHydrantFormData, "isValid">;
+
+interface IValidation {
+  name: string;
+  isValid: (formData: IHydrantFormData) => boolean;
+  message?: IValidationMessage;
+}
+
+type IFormValidations = Record<
+  IFormValidationKey,
+  { validations: IValidation[] }
```

Comment:
As implemented here the order of these validations defines their order of
precedence when validating the form which makes great sense. When making this a generalized abstraction, can we have a
comment making that explicit?